tools/manifest/manifest.py

```diff
@@ -306,10 +312,7 @@ def update(self, tree):
                     _, old_type = self._path_hash[rel_path]
                     if old_type in reftest_types:
                         reftest_changes = True
-                    try:
-                        del self._path_hash[rel_path]
-                    except KeyError:
-                        pass
+                    del self._path_hash[rel_path]
```

Comment:
IIRC, this handles the `KeyError` for when we have an old manifest without the hash data, but maybe we just outright fail without that nowadays.

Reply:
We already accessed `self._path_hash[rel_path]` a few lines above without any exception handling, so if it doesn't exist, we wouldn't even get to this point.